### PR TITLE
fix(wallet): Ledger transaction submission bug fix

### PR DESCRIPTION
**Pull request by tomislavhoracek**
# Context
Bugfix for transaction signing with Ledger devices when the account index is different than zero.

# Proposed Solution
[JIRA ticket](https://input-output.atlassian.net/browse/ADP-1800)
Wallet user can create a single account / address wallet with any account index, and perform all wallet actions using it. In this particular case to sign and send transactions using Ledger.

# Important Changes Introduced
- Fixed witnesses preparation in LedgerKeyAgent

### CODE REVIEW

**Review by mkazlauskas — APPROVED** [review]

**Review by rhyslbw — CHANGES_REQUESTED** [review]
From the bug report, the expected behavior is:

> Wallet users can create a single account/address wallet with any account index, and perform all wallet actions using it. In this particular case to sign and send transactions using Ledger

Please add test coverage for this to demonstrate we're meeting the requriements.

**Review by rhyslbw — DISMISSED** [review]

File: packages/wallet/src/KeyManagement/LedgerKeyAgent.ts [hunk 1]
```diff
@@ -243,7 +243,7 @@ export class LedgerKeyAgent extends KeyAgentBase {
         await Promise.all(
           result.witnesses.map(async (witness) => {
             const publicKey = await this.derivePublicKey({
-              index: HARDENED - witness.path[2],
+              index: witness.path[4],
```
[the remark this answers is not in the document]
Comment (tomislavhoracek) [reply packages/wallet/src/KeyManagement/LedgerKeyAgent.ts:246]: Thanks @rhyslbw. I was thinking to introduce that in the scope of another PR where we introduced constants ( purpose, coin_type)
https://github.com/input-output-hk/cardano-js-sdk/blob/a7cbf49b2445eeeaa4fbae4100925f37c3d0989e/packages/wallet/src/KeyManagement/util/constants.ts#L1

[the remark this answers is not in the document]
Comment (rhyslbw) [reply packages/wallet/src/KeyManagement/LedgerKeyAgent.ts:246]: Great, I'll retract my CR.

**Review by rhyslbw — CHANGES_REQUESTED** [review]
Looks good, please squash that last commit in preparation for the final QA review

**Review by mrcnk — APPROVED** [review]
I ran the QA process for this bug fix on the frontend side and everything seems to be working correctly (when creating a wallet with `accountIndex` different than the first one).

**Review by rhyslbw — APPROVED** [review]

**Pull request merged**